app/main.py

```diff
@@ -1,2 +1,133 @@
+        self._capacity: int = INITIAL_CAPACITY
+        self._hash_table: list = [None] * self._capacity
+
+    def __str__(self) -> str:
+        items = []
+        for node in self._hash_table:
+            if node is not None:
+                items.append(f"'{node.key}': {node.value}")
+        return "{" + ", ".join(items) + "}"
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        node = Node(key, value)
+        index = node.hash_key % self._capacity
+
+        while (self._hash_table[index] is not None
+               and self._hash_table[index].key != key):
+            index = (index + 1) % self._capacity
+
+        if self._hash_table[index] is None:
+            self._length += 1
+
+        self._hash_table[index] = node
+
+        if self._length / self._capacity > LOAD_FACTOR:
+            self._resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        key_hash = hash(key)
+        index = key_hash % self._capacity
+
+        while (self._hash_table[index] is not None
+               and self._hash_table[index].key != key):
+            index = (index + 1) % self._capacity
+
+        if self._hash_table[index] is None:
+            raise KeyError(key)
+
+        return self._hash_table[index].value
+
+    def __len__(self) -> int:
+        return self._length
+
+    def __delitem__(self, key: Hashable) -> None:
+        key_hash = hash(key)
+        index = key_hash % self._capacity
+
+        while (self._hash_table[index] is not None
+               and self._hash_table[index].key != key):
+            index = (index + 1) % self._capacity
+
+        if self._hash_table[index] is None:
+            raise KeyError(key)
+
+        self._hash_table[index] = None
+        self._length -= 1
+
+    def keys(self) -> Any:
+        for i in range(len(self._hash_table)):
+            if self._hash_table[i] is not None:
+                self._dict_keys.append(self._hash_table[i].key)
+        return self._dict_keys
+
+    def values(self) -> Any:
+        for i in range(len(self._hash_table)):
+            if self._hash_table[i] is not None:
+                self._dict_values.append(self._hash_table[i].value)
+        return self._dict_values
+
+    def items(self) -> list:
+        return [(self._hash_table[i].key, self._hash_table[i].value) for i in
+                range(len(self._hash_table)) if
+                self._hash_table[i] is not None]
+
+    def clear(self) -> None:
+        self._hash_table = []
+        self._dict_keys = []
+        self._dict_values = []
+        self._length = 0
+
+    def get(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            self.__getitem__(key)
+        except KeyError:
+            return default
+
+    def pop(self, _key: Any, default: Any = None) -> Any:
+        try:
+            key_value = self[_key]
+            self.__delitem__(key_value)
+            return key_value
+        except KeyError:
+            return default
```

Comment:
It returns default if `default` is provided. In other cases `KeyError` is raised

Reply:
Fixed